seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/util/ClickhouseCatalogUtil.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.clickhouse.util;
+
+import org.apache.seatunnel.api.sink.SaveModePlaceHolder;
+import org.apache.seatunnel.api.table.catalog.Column;
+import org.apache.seatunnel.api.table.catalog.TablePath;
+import org.apache.seatunnel.api.table.catalog.TableSchema;
+import org.apache.seatunnel.connectors.seatunnel.clickhouse.catalog.ClickhouseTypeConverter;
+import org.apache.seatunnel.connectors.seatunnel.clickhouse.config.ClickhouseConfig;
+import org.apache.seatunnel.connectors.seatunnel.common.sql.template.SqlTemplate;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class ClickhouseCatalogUtil {
+
+    public static String getCreateTableSql(
+            String template, String database, String table, TableSchema tableSchema) {
+        String primaryKey = "";
+        if (tableSchema.getPrimaryKey() != null) {
+            primaryKey =
+                    tableSchema.getPrimaryKey().getColumnNames().stream()
+                            .map(r -> "`" + r + "`")
+                            .collect(Collectors.joining(","));
+        }
+        String uniqueKey = "";
+        if (!tableSchema.getConstraintKeys().isEmpty()) {
+            uniqueKey =
+                    tableSchema.getConstraintKeys().stream()
+                            .flatMap(c -> c.getColumnNames().stream())
+                            .map(r -> "`" + r.getColumnName() + "`")
+                            .collect(Collectors.joining(","));
+        }
+        SqlTemplate.canHandledByTemplateWithPlaceholder(
+                template,
+                SaveModePlaceHolder.ROWTYPE_PRIMARY_KEY.getPlaceHolder(),
+                primaryKey,
+                TablePath.of(database, table).getFullName(),
+                ClickhouseConfig.SAVE_MODE_CREATE_TEMPLATE.key());
+        template =
+                template.replaceAll(
```

Comment:
how about put template parse into common module? Not only one connector use same logic.